compute/cloud-client/src/test/java/compute/Util.java

```diff
@@ -195,15 +201,71 @@ public static void cleanUpExistingReservations(
         if (!reservationsClient.list(projectId, zone).iterateAll().iterator().hasNext()) {
           break;
         }
-        if (reservation.getName().contains(prefixToDelete)
+        if (containPrefixToDelete(reservation, prefixToDelete)
             && isCreatedBeforeThresholdTime(reservation.getCreationTimestamp())) {
-          try {
-            DeleteReservation.deleteReservation(projectId, zone, reservation.getName());
-          } catch (NotFoundException e) {
-            System.err.println("Reservation not found, skipping deletion:" + reservation.getName());
-          }
+          DeleteReservation.deleteReservation(projectId, zone, reservation.getName());
         }
       }
     }
   }
+
+  // Delete disks which starts with the given prefixToDelete and
+  // has creation timestamp >24 hours.
+  public static void cleanUpExistingDisks(String prefixToDelete, String projectId,
+                                          String zone)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    try (DisksClient disksClient = DisksClient.create()) {
+      for (Disk disk : disksClient.list(projectId, zone).iterateAll()) {
+        if (!disksClient.list(projectId, zone).iterateAll().iterator().hasNext()) {
+          break;
+        }
+        if (containPrefixToDelete(disk, prefixToDelete)
+            && isCreatedBeforeThresholdTime(disk.getCreationTimestamp())) {
+          DeleteDisk.deleteDisk(projectId, zone, disk.getName());
+        }
+      }
+    }
+  }
+
+  // Delete snapshots which starts with the given prefixToDelete and
+  // has creation timestamp >24 hours.
+  public static void cleanUpExistingSnapshots(String prefixToDelete, String projectId)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    try (SnapshotsClient snapshotsClient = SnapshotsClient.create()) {
+      for (Snapshot snapshot : snapshotsClient.list(projectId).iterateAll()) {
+        if (!snapshotsClient.list(projectId).iterateAll().iterator().hasNext()) {
+          break;
+        }
+        if (containPrefixToDelete(snapshot, prefixToDelete)
+            && isCreatedBeforeThresholdTime(snapshot.getCreationTimestamp())) {
+          DeleteSnapshot.deleteSnapshot(projectId, snapshot.getName());
+        }
+      }
+    }
+  }
+
+  public static boolean containPrefixToDelete(
```

Comment:
Added to check if resource has prefixToDelete and to prevent error on resource.getName()  call